foreign_cc/private/cmake_script.bzl

```diff
@@ -93,13 +116,13 @@ def create_cmake_script(
     if not params.cache.get("CMAKE_RANLIB"):
         params.cache.update({"CMAKE_RANLIB": ""})
 
-    # Avoid cmake passing wrong linker flags when targeting android on macOS
-    # https://github.com/bazelbuild/rules_foreign_cc/issues/289
-    if target_os == "android":
-        params.cache.update({
-            "ANDROID": "YES",
-            "CMAKE_SYSTEM_NAME": "Linux",
-        })
+    # Avoid CMake passing the wrong linker flags when cross compiling
+    # by setting CMAKE_SYSTEM_NAME and CMAKE_SYSTEM_PROCESSOR,
+    # see https://github.com/bazelbuild/rules_foreign_cc/issues/289,
+    # and https://github.com/bazelbuild/rules_foreign_cc/pull/1062
+    if target_os != host_os and target_os != "unknown":
```

Comment:
There's more to it than this (i.e. linux-x86_64 -> linux-aarch64).

Probably should be a function like `is_cross_compilation` and handle this more rigorously.